src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -816,10 +854,11 @@ def update_service(
 
         # Go into the update state
         try:
-            service.state = service_entity.lifecycle.get_transfer(
+            transfer = service_entity.lifecycle.get_transfer(
                 from_state=service.state,
                 transfer_type=inmanta_lsm.const.TransferTrigger.ON_UPDATE,
-            ).target
```

Comment:
The current implementation is not entirely correct, but I have the feeling that the update path was not correct before. The `version` field was not updated, and we were not doing a recompile, am I missing something?

Reply:
We were doing all of that in the auto_transfer method, but you changed the logic to compile after the transfer, instead of compiling before the transfer, this is very different and indeed breaks everything